android/src/main/kotlin/com/shopify/reactnative/flash_list/AutoLayoutShadow.kt

```diff
@@ -25,23 +25,27 @@ class AutoLayoutShadow {
         for (i in 0 until sortedItems.size - 1) {
             val cell = sortedItems[i]
             val neighbour = sortedItems[i + 1]
+            // Only apply correction if the next cell is consecutive.
+            val isNeighbourConsecutive = cell.index == neighbour.index + 1
```

Comment:
Should we apply this logic on iOS, too, or does the problem not manifest there?

Reply:
iOS changes are also present in this PR. Check below